java/core/src/java/org/apache/orc/impl/OrcTail.java

```diff
@@ -121,8 +121,10 @@ public OrcProto.PostScript getPostScript() {
   public OrcFile.WriterVersion getWriterVersion() {
     OrcProto.PostScript ps = fileTail.getPostscript();
     OrcProto.Footer footer = fileTail.getFooter();
-    OrcFile.WriterImplementation writer =
-        OrcFile.WriterImplementation.from(footer.getWriter());
+    OrcFile.WriterImplementation writer = OrcFile.WriterImplementation.UNKNOWN;
+    if (footer.hasWriter()) {
+      writer = OrcFile.WriterImplementation.from(footer.getWriter());
+    }
```

Comment:
Technically, this read-path change doesn't match with our writer behavior because we throws `IllegalArgumentException` explicitly.

https://github.com/apache/orc/blob/873d00c1f42753e9522cc488db59b1c979299eea/java/core/src/java/org/apache/orc/OrcFile.java#L749-L752

Reply:
I don't think so; A low version writer certainly can't specify that the orc written is a futrue version. But a reader of a low version should be able to read an orc file created by a higher version, even if that higher version is unknown in the low-code implementation.

Reply:
No, it seems that you are confused with the literal `FUTURE` version. The literal `FUTURE` version means `infinite` in Math, @guiyanakuang .

Reply:
I believe UNSTABLE_PRE_2_0 and FUTURE are for developer testing, and not valid for the production ORC files.
https://github.com/apache/orc/blob/873d00c1f42753e9522cc488db59b1c979299eea/java/core/src/java/org/apache/orc/OrcFile.java#L71-L85

Reply:
That being the case, close this PR and keep the status quo